crates/hir-expand/src/fixup.rs

```diff
@@ -488,6 +484,7 @@ fn foo () {a .__ra_fixup ;}
     }
 
     #[test]
+    #[ignore]
```

Comment:
These two tests fail in [this assertion](https://github.com/rust-lang/rust-analyzer/blob/df3877037e448ca90b17525b00418a2c30da8ddb/crates/hir-expand/src/fixup.rs#L351-L354). This is because dummy ident insertion (to fix up the syntax error) changes the spacing of puncts before it with this patch. Although technically we can record and reverse it to preserve the spacing of the puncts in invalid syntax, it'd be somewhat complex and I'm wondering how essential this invariant to preserve the original tokentree is w.r.t. puncts' spacing.

Reply:
Hm, it seems odd to me that we check if *spacing* is preserved, macros already don't care about spacing so I don't think the tests should either, not sure how simple it'd would be but can we just check the tokens for equality ignoring any whitespace tokens in between?

Reply:
Yeah that's what I was thinking. I'll see how I can "fix" the tests without complicating much.

cc @flodiebold for you're the original author (#11444), if you have anything to add.